### PR TITLE
Scanning simulated peripherals

### LINKED ISSUE #12: Handle scanning
When user starts scanning for peripherals, real data must be returned every advertisement interval.
Call to stop scan must be implemented and handled.
----------
RxJS/RxTs? Callbacks? Some other streams?

### PR DESCRIPTION
**Pull request by mikolak**
Resolves #12

### CODE REVIEW

File: src/blemulator.ts [hunk 1]
```diff
@@ -73,10 +66,10 @@ class BlemulatorInstance {
 
     private test(callbackId: String) {
```
Comment (mikolak) [inline src/blemulator.ts:67]: Should be removed

File: android/src/main/java/com/polidea/blemulator/PlatformToJsBridge.java [hunk 2]
```diff
@@ -72,8 +47,7 @@ public void startScan(String[] filteredUUIDs,
                     @Override
                     public void invoke(ReadableMap args) {
                         if (args.hasKey(NativeArgumentName.ERROR)) {
-                            //TODO proper error handling (#12)
-                            onErrorCallback.onError(new BleError(BleErrorCode.UnknownError, "", -1));
+                            onErrorCallback.onError(parseError(args.getMap(NativeArgumentName.ERROR)));
```
[the remark this answers is not in the document]
Comment (mikolak) [reply android/src/main/java/com/polidea/blemulator/PlatformToJsBridge.java:50]: It's working.

File: src/internal/simulation-manager.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    private filteredUuids?: Array<UUID> = undefined
+    private advertisementIntervalHandles: Array<number> = []
+    private isScanInProgress: boolean = false
+
+    addPeripheral(peripheral: SimulatedPeripheral): void {
+        this.peripherals.push(peripheral)
+        if (this.isScanInProgress) {
+            if (this.shouldAdvertise(peripheral, this.filteredUuids))
+                this.setAdvertisement(peripheral)
+            this.setAdvertisement(peripheral)
```
Comment (Cierpliwy) [inline src/internal/simulation-manager.ts:26]: Why we call it twice sometimes?

File: src/internal/simulation-manager.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        this.addScanResult = addScanResult
+        this.filteredUuids = filteredUuids
+        this.peripherals.forEach((peripheral) => {
+            if (this.shouldAdvertise(peripheral, this.filteredUuids))
+                this.setAdvertisement(peripheral)
+        })
+        this.isScanInProgress = true
+
+        return undefined
```
Comment (Cierpliwy) [inline src/internal/simulation-manager.ts:53]: I think function returns undefined by default.

File: src/internal/simulation-manager.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        this.peripherals.push(peripheral)
+        if (this.isScanInProgress) {
+            if (this.shouldAdvertise(peripheral, this.filteredUuids))
+                this.setAdvertisement(peripheral)
+            this.setAdvertisement(peripheral)
+        }
+    }
+
+    startScan(filteredUuids: Array<UUID> | undefined, scanMode: number | undefined,
+        callbackType: number | undefined, addScanResult: ScanResultListener): SimulatedBleError | undefined {
```
Comment (Cierpliwy) [inline src/internal/simulation-manager.ts:31]: Is there any advantage to use `undefined` instead of `null`?

Comment (mikolak) [reply src/internal/simulation-manager.ts:31]: Yeah, undefined is default if something's not passed, so you can just omit it. If you make it nullable, you'll have to pass the null each time - _I think_.

File: src/internal/simulation-manager.ts (hunk 3 above)
Comment (mikolak) [reply src/internal/simulation-manager.ts:26]: My bad. :) Good catch!

File: src/internal/simulation-manager.ts (hunk 4 above)
Comment (mikolak) [reply src/internal/simulation-manager.ts:53]: So it would seem. Removing the line.

**Pull request merged**